rust/otap-dataflow/crates/syslog_cef_receiver/src/arrow_records_encoder.rs

```diff
@@ -0,0 +1,2963 @@
+    curr_log_id: u16,
+    logs: LogsRecordBatchBuilder,
+    log_attrs: AttributesRecordBatchBuilder<u16>,
+}
+
+impl ArrowRecordsBuilder {
+    #[must_use]
+    pub(crate) fn new() -> Self {
+        Self {
+            curr_log_id: 0,
+            logs: LogsRecordBatchBuilder::new(),
+            log_attrs: AttributesRecordBatchBuilder::<u16>::new(),
+        }
+    }
+
+    pub(crate) fn len(&self) -> u16 {
+        // Current log record ID is incremented for each new log appended
+        // so it can be used to get the number of logs in the builder.
+        self.curr_log_id
+    }
+
+    pub(crate) fn append_syslog(&mut self, syslog_message: ParsedSyslogMessage<'_>) {
+        self.logs
+            .append_time_unix_nano(syslog_message.timestamp().map(|v| v as i64));
+
+        let (severity_number, severity_text) =
+            syslog_message.severity().unwrap_or((0, "UNSPECIFIED"));
+        self.logs.append_severity_number(Some(severity_number));
+        self.logs.append_severity_text(Some(severity_text));
+
+        self.logs.body.append_str(syslog_message.input().as_ref());
+
+        let attributes_added = syslog_message.add_attribues_to_arrow(&mut self.log_attrs);
+
+        for _ in 0..attributes_added {
+            self.log_attrs.append_parent_id(&self.curr_log_id);
+        }
+
+        self.logs.append_id(Some(self.curr_log_id));
+
+        self.curr_log_id += 1;
+    }
+
+    pub(crate) fn build(mut self) -> Result<OtapArrowRecords> {
+        let log_record_count = self.curr_log_id.into();
+
+        // All the logs are belong the same resource and scope
+        self.logs.resource.append_id_n(0, log_record_count);
+        self.logs
+            .resource
+            .append_schema_url_n(None, log_record_count);
+        self.logs
+            .resource
+            .append_dropped_attributes_count_n(0, log_record_count);
+
+        self.logs.scope.append_id_n(0, log_record_count);
+        self.logs.scope.append_name_n(None, log_record_count);
+        self.logs.scope.append_version_n(None, log_record_count);
+        self.logs
+            .scope
+            .append_dropped_attributes_count_n(0, log_record_count);
+
+        let observed_time = Utc::now().timestamp_nanos_opt().unwrap_or(0);
+        for _ in 0..log_record_count {
+            self.logs
+                .append_observed_time_unix_nano(Some(observed_time));
+        }
+
+        self.logs.append_schema_url_n(None, log_record_count);
+
+        for _ in 0..log_record_count {
+            self.logs.append_dropped_attributes_count(0);
+        }
+
+        for _ in 0..log_record_count {
+            self.logs.append_flags(None);
+        }
+
+        for _ in 0..log_record_count {
+            _ = self.logs.append_trace_id(None);
+        }
+
+        for _logs in 0..log_record_count {
+            _ = self.logs.append_trace_id(None);
+        }
+
+        for _logs in 0..log_record_count {
+            _ = self.logs.append_span_id(None);
+        }
```

Comment:
Not completely familiar with the latest version of the API, but couldn’t we call an `append_null_n` or something similar here?